### ISSUE TITLE
对于代码一些疑问

### ISSUE THREAD

**Issue by yysirs** [beyondguo/label_confusion_learning#11]
实现LCM_loss的时候，使用两个loss1，loss2。
`loss1 = -K.categorical_crossentropy(simulated_y_true, simulated_y_true)`
对于loss1的作用能不能解释一下，在论文里也没有提及。

**Comment by beyondguo** [comment]
看看其他的issue，解释过了。看来这里确实很容易产生疑问🤔️

**Comment by yysirs** [comment]
框架我使用的torch，在forward中，我将one-hot的label转为embedding。shape为10*3*100 (batch_size,label_num,hidden_size).
但是经过lstm的out为10*3，对于dot product计算这里 。应该怎么转变维度。
我的代码如下：
`class BiLSTM_Model_LCM(nn.Module):
    """
    参考论文《Label Confusion Learning to Enhance Text Classification Models》
    """
    def __init__(self, vocab_size, label_size, static_embeddings):
        super(BiLSTM_Model_LCM, self).__init__()
        self.hidden_dim = config.hidden_size
        self.embedding_dim = config.embedding_dim
        self.vocab_size = vocab_size
        self.label_size = label_size
        self.embedding = nn.Embedding(self.vocab_size, self.embedding_dim)
        # 这里使用static_embeddings来初始化embedding
        self.embedding.weight.data.copy_(torch.from_numpy(static_embeddings))
        self.BiLstm = nn.LSTM(self.embedding_dim, self.hidden_dim, num_layers=1, bidirectional=True, batch_first=True)
        self.dropout = nn.Dropout(config.dropout)
        self.linear1 = nn.Linear(self.hidden_dim*2, self.hidden_dim)
        self.linear2 = nn.Linear(self.hidden_dim, self.label_size)
        self.tanh = nn.Tanh()
        # self.mm = torch.mm()

    def forward(self, input, label):
        # 句子的embeding
        print(label.shape)
        input_embedding = self.embedding(input)
        '''预测句子的标签'''
        output, (h_n, c_n) = self.BiLstm(input_embedding)
        # 拼接正向最后一个输出和反向最后一个输出
        inputs = [h_n[-1, :, :], c_n[-2, :, :]]
        out = torch.cat(inputs, dim=-1)
        # 进行全连接
        out_fc1 = self.linear1(out)
        # 进行relu
        out_fc1_relu = F.relu(out_fc1)
        # 全连接
        out = self.linear2(out_fc1_relu) #shape(batch_size, label_num)
        # 句子预测的伪标签
        pred_probs = F.log_softmax(out, dim=-1)#shape(batch_size, label_num)
        print('pred_probs shape=', pred_probs)
        # label的embedding
        label_embedding = self.embedding(label)##shape(batch_size, label_num,hidden_size)
        # 激活函数
        label_embedding = self.tanh(label_embedding)
        # similarity part:
        print('out shape=', out.shape)
        print('label_embedding shape=', label_embedding.shape)
        dot_product = torch.mm(label_embedding, out)
        # 标签和输入文本的相似度得到的标签混淆分布
        label_sim_dict = F.softmax(dot_product, dim=-1)
        # concat output:
        out_puts = [pred_probs[-1, :], label_sim_dict[-2, :]]
        cat_out = torch.cat(out_puts, dim=-1)
        return cat_out`

**Comment by Anastasiawang** [comment]
哈喽我也是正在尝试在pytorch框架基础上复现这一方法，也是遇到一点问题，能交流一下吗？

**Comment by Josoope** [comment]
请问转成pytorch框架了吗？
